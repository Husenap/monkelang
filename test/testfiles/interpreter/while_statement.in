monke has number bananas = 0

while bananas <= 10 monke do
  monke think bananas > 5 monke do
    monke say bananas + " bananas is too many bananas!"
  monke think again bananas == 5 monke do
    monke say "perfect amount of bananas 😍"
  monke change mind
    monke say "only " + bananas + " bananas, need more bananas"
  monke thinked

  bananas = bananas + 1
monke done

