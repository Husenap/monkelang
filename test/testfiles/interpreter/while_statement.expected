only 0 bananas, need more bananas
only 1 bananas, need more bananas
only 2 bananas, need more bananas
only 3 bananas, need more bananas
only 4 bananas, need more bananas
perfect amount of bananas 😍
6 bananas is too many bananas!
7 bananas is too many bananas!
8 bananas is too many bananas!
9 bananas is too many bananas!
10 bananas is too many bananas!
